src/dataExplorer/components/ResultsPane.tsx

```diff
@@ -126,8 +127,17 @@ const ResultsPane: FC = () => {
       let inputValue: string = ''
 
       switch (language) {
-        case LanguageType.FLUX:
         case LanguageType.SQL:
+          if (isFlagEnabled('v2privateQueryUI')) {
+            url = `${API_BASE_PATH}api/v2private/query?database=${selection.bucket?.name}`
+            // the `\n` character is interpreted by the browser as a line break
+            // so replacing the `\n` char with empty space to keep query in correct syntax
+            inputValue = text.replace(/\n/g, ' ')
+            break
```

Comment:
I think this `break` needs to be outside the `if` block. If the language type is SQL (the `case`) but the underlying `if` block with the feature flag isn't satisfied, javascript will miss the `break` and continue to the remaining case statements (and most importantly, to the `default` block that throws an error) https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch#breaking_and_fall-through

Reply:
The break is intentionally put inside the `if` statement. When the feature flag is not satisfied, then SQL falls to the next case which is the Flux case. 

When the feature flag `v2privateQueryUI` is off,  each SQL query actually is wrapped inside a Flux query `iox.sql()`. 

Sorry it is kind of confusing 😅 I'll add a comment to make it more clear

Reply:
Updated comments in https://github.com/influxdata/ui/pull/6792/commits/b384bf99036faa6dde6290706e0b43170e59b527

Reply:
Got it. Thanks for explaining the logic and updating this comment 👍